src/ResourceManagement/DataLake.Store/DataLakeStore.Tests/DataTransferTests/MultipleSegmentUploaderTests.cs

```diff
@@ -274,40 +274,40 @@ private void VerifyTargetStreamIsComplete(UploadSegmentMetadata segmentMetadata,
         }
 
 
-        private byte[] GetExpectedContents(UploadSegmentMetadata segment, UploadMetadata metadata)
+        private byte[] GetExpectedContents(TransferSegmentMetadata segment, TransferMetadata metadata)
         {
             byte[] result = new byte[segment.Length];
-            Array.Copy(_smallFileContents, segment.SegmentNumber * metadata.SegmentLength, result, 0, segment.Length);
+            Array.Copy(_smallFileContents, (int)(segment.SegmentNumber * metadata.SegmentLength), result, 0, (int)segment.Length);
```

Comment:
This is required because segmentLength is a long and you must explicitly down cast. There is no risk for loss of precision here, though, since the test ensures a very small file size.